README.md

```diff
@@ -1,9 +1,111 @@
 # Ultimate TicTacToe Backend
 
 [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/ultimate-ttt/ultimate-ttt-backend/issues)
+[![Build Status](https://dev.azure.com/ultimate-ttt/ultimate-ttt/_apis/build/status/ultimate-ttt-Build-master?branchName=master)](https://dev.azure.com/ultimate-ttt/ultimate-ttt/_build/latest?definitionId=4&branchName=master)
+[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=ultimate-ttt-backend&metric=alert_status)](https://sonarcloud.io/dashboard?id=ultimate-ttt-backend)
+[![Coverage](https://sonarcloud.io/api/project_badges/measure?project=ultimate-ttt-backend&metric=coverage)](https://sonarcloud.io/dashboard?id=ultimate-ttt-backend)
+[![GitHub release](https://img.shields.io/github/release/ultimate-ttt/ultimate-ttt-backend.svg)](https://github.com/ultimate-ttt/ultimate-ttt-backend/releases)
+[![Docker Pulls](https://img.shields.io/docker/pulls/ultimatettt/ultimate-ttt-server.svg)](https://hub.docker.com/r/ultimatettt/ultimate-ttt-server)
 
-# Local Testing
-TODO
+# Public API
+TODO Visit [#10](https://github.com/ultimate-ttt/ultimate-ttt-backend/issues/10) to see Progress
+
+# Running the Api on your system
+
+## Using the latest release of the Docker image
+
+To use the latest release for local testing run the following steps. dotnet doesn't need to be installed on your local computer.
+
+**Prerequisites**
+- Docker is installed
+
+**Setup**
+
+1. Open command-line and navigate to the root of the repository
+2. run `docker-compose up`
+3. Access the API from your browser [http://localhost:5023/playground](http://localhost:5023/playground)
+
+## Work on the code
+
+If you work on the code you probably want to run the Api and check if your changes work as expected.
+
+**Prerequisites**
+- dotnet core 2.1 SDK installed
+  
+**Build and Test the Api**
+
+1. Open command-line and navigate to the root of the repository
+2. run `build .ps1` on a windows system or `build.sh` on linux
```

Comment:
I think you need `.\build.ps1` otherwise I got this error:
```
The command build.ps1 was not found, but does exist in the current location. PowerShell does not load commands from the current location by default. If you trust this command, instead type: ".\build.ps1". See "get-help about_Command_Precedence" for more details.
```
(granted I ran this in https://github.com/PowerShell/PowerShell, maybe the behaviour is different on the normal Windows PowerShell?)

Reply:
Then ran into the issue that this requires Cake.exe 😄 
So I tried with build.sh but I have trouble to get this working:
```
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ build.sh
-bash: build.sh: command not found
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ sh build.sh
build.sh: line 74: mono: command not found
Could not restore NuGet tools.
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ chmod +x build.sh
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ build.sh
-bash: build.sh: command not found
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ ./build.sh
./build.sh: line 74: mono: command not found
Could not restore NuGet tools.
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ 
```

Reply:
yes on powershell it should be .\build.ps1. I will change that

Reply:
Maybe you can figure it out how to get it running.

https://cakebuild.net/docs/tutorials/setting-up-a-new-project

Reply:
So it seems like you need to have mono installed to use the build.sh (wow..)
But now I actually run into the same issue as with powershell:

sh:
```
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ sh build.sh
Feeds used:
  /Users/Maurice/.nuget/packages/
  https://api.nuget.org/v3/index.json

Restoring NuGet package Cake.0.33.0.
  GET https://api.nuget.org/v3-flatcontainer/cake/0.33.0/cake.0.33.0.nupkg
  OK https://api.nuget.org/v3-flatcontainer/cake/0.33.0/cake.0.33.0.nupkg 22ms
Installing Cake 0.33.0.
Adding package 'Cake.0.33.0' to folder '/Users/Maurice/Development/backend-ultimate-ttt/.packages'
Added package 'Cake.0.33.0' to folder '/Users/Maurice/Development/backend-ultimate-ttt/.packages'
Could not find Cake.exe at '/Users/Maurice/Development/backend-ultimate-ttt/tools/Cake/Cake.exe'.
```

Powershell:
```
Could not find Cake.exe at /Users/Maurice/Development/backend-ultimate-ttt/tools/Cake/Cake.exe
At /Users/Maurice/Development/backend-ultimate-ttt/build.ps1:225 char:5
+     Throw "Could not find Cake.exe at $CAKE_EXE"
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (Could not find ...s/Cake/Cake.exe:String) [], RuntimeException
+ FullyQualifiedErrorId : Could not find Cake.exe at /Users/Maurice/Development/backend-ultimate-ttt/tools/Cake/Cake.exe
```

Do you have a .exe there on your computer?

Reply:
No but windows can execute exes.

Reply:
Mac can aswell. (cake uses nuget.exe / dotnet.exe)
Would need to check again but I don't think so. I don't think this has anything to do with that anyway?!

Reply:
@maracuja-juice Can you try if it works with the CLI tool as described in a blog by them.

[https://cakebuild.net/blog/2018/08/cake-v0.30.0-released](https://cakebuild.net/blog/2018/08/cake-v0.30.0-released)

Reply:
Do you mean like this?
```
dotnet tool install -g Cake.Tool --version 0.30.0
dotnet cake build.cake
```
Output:
```
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ dotnet cake build.cake
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(17,15): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(19,18): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(20,18): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(21,19): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
Error: Error(s) occurred when compiling build script:
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(17,15): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(19,18): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(20,18): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
/Users/Maurice/Development/backend-ultimate-ttt/build.cake(21,19): error CS0308: The non-generic method 'EnvironmentVariable(string)' cannot be used with type arguments
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ 
```

Maybe you should spin up a Linux VM and try to get it working there.

Reply:
Sure you have cake 0.33.0?

Reply:
Ah I see. Yep this works. 
Now the tests fail :)

```
[xUnit.net 00:00:02.09]     Domain.Tests.GameManagerTests.Move_ValidMove_CorrectResult [FAIL]
Failed   Domain.Tests.GameManagerTests.Move_ValidMove_CorrectResult
Error Message:
 Snapshooter.Exceptions.SnapshotTestException : Could not resolve directory for SnapShot
Stack Trace:
   at Snapshooter.Xunit.LiveUnitTestingDirectoryResolver.CheckForSession(SnapshotFullName snapshotFullName) in /home/vsts/work/1/s/src/Snapshooter.Xunit/LiveUnitTestingDirectoryResolver.cs:line 50
   at Snapshooter.Xunit.XunitSnapshotFullNameReader.ReadSnapshotFullName() in /home/vsts/work/1/s/src/Snapshooter.Xunit/XunitSnapshotFullNameReader.cs:line 52
   at Snapshooter.SnapshotFullNameResolver.ResolveSnapshotFullName(String snapshotName, String snapshotNameExtension) in /home/vsts/work/1/s/src/Snapshooter/SnapshotFullNameResolver.cs:line 69
   at Snapshooter.Xunit.Snapshot.Match(Object currentResult, Func`2 matchOptions) in /home/vsts/work/1/s/src/Snapshooter.Xunit/Snapshot.cs:line 145
   at Domain.Tests.GameManagerTests.Move_ValidMove_CorrectResult()
--- End of stack trace from previous location where exception was thrown ---
Results File: /Users/Maurice/Development/backend-ultimate-ttt/testoutput/_Marimbas-MacBook-Pro_2019-04-25_20_45_25.trx

Total tests: 10. Passed: 9. Failed: 1. Skipped: 0.
Test Run Failed.
```

Reply:
So we can write that down that you only need dotnet + this tool now to get it (almost) working. 
You don't need mono (I uninstalled in the meantime again)

Reply:
👍 The test is working if you run in debug. Release mode tests are kind of buggy with Snapshooter.

`dotnet cake build.cake -target "Debug"` should work

I will update the dokumentation.

Reply:
1. You're right the tests work when running this in the Rider text explorer for example
2. What am I doing wrong?
```
[Maurice@Marimbas-MacBook-Pro ~/Development/backend-ultimate-ttt (readme)]$ dotnet cake build.cake -target "Debug"
More than one build script specified.
```
3. I opened an issue for this in the Snapshooter repo: https://github.com/SwissLife-OSS/snapshooter/issues/50

Reply:
Why did you delete my comment?? 😲

Reply:
Was an accident😬 can't restore sorry